allenact_plugins/manipulathor_plugin/manipulathor_environment.py

```diff
@@ -258,27 +324,88 @@ def close_enough(self, current_obj_pose, init_obj_pose, threshold):
             for k in ["x", "y", "z"]
         ]
         position_is_close = sum(position_close) == 3
-        rotation_close = [
-            abs(current_obj_pose["rotation"][k] - init_obj_pose["rotation"][k])
-            <= threshold
-            for k in ["x", "y", "z"]
-        ]
-        rotation_is_close = sum(rotation_close) == 3
-        return position_is_close and rotation_is_close
-
-    def get_objects_moved(self, initial_object_locations):
-        current_object_locations = self.get_current_object_locations()
+        return position_is_close
+
+        # def deg_dist(d0: float, d1: float):
+        #     dist = (d0 - d1) % 360
+        #     return min(dist, 360 - dist)
+
+        # rotation_close = [
+        #     deg_dist(current_obj_pose["rotation"][k], init_obj_pose["rotation"][k])
+        #     <= threshold
+        #     for k in ["x", "y", "z"]
+        # ]
+        # rotation_is_close = sum(rotation_close) == 3
+        # return position_is_close and rotation_is_close
+
+    def get_objects_moved(
+        self, previous_object_locations, current_object_locations, target_object_id,
+    ):
         moved_objects = []
+        scene_id = self.scene_name.split("_")[0]
+
         for object_id in current_object_locations.keys():
+            if object_id == target_object_id:
+                continue
+            if object_id not in previous_object_locations:
+                continue
+
+            thres = self.vibration_distances[scene_id + "-" + object_id]
+
             if not self.close_enough(
                 current_object_locations[object_id],
-                initial_object_locations[object_id],
-                threshold=UNWANTED_MOVE_THR,
+                previous_object_locations[object_id],
+                threshold=max(UNWANTED_MOVE_THR, thres),
```

Comment:
`thres` should be a parameter to this function with default value `UNWANTED_MOVE_THR`. Anyone who calls this can then use the vibration distances if they want to (i.e. using these distances should be handled at the task level).

Reply:
I'm not clear why we have two values, three and UNWANTED_MOVE_THR. I think it should be only one of them.

Reply:
Since object can be vibrated by default, so `UNWANTED_MOVE_THR = 0.01` maybe not large enough. Nevertheless, I agree it should be moved to custom repo.